New: Add division element to neumes module

fujinaga opened the pull request:
The division element is currently missing from the neumes module. These symbols look like barlines but have a function similar to breath marks and rests, with no specific duration.

The lilypond documentation gives some examples of divisions: http://lilypond.org/doc/v2.18/Documentation/notation/typesetting-gregorian-chant#divisiones

They are also available as SMuFL symbols here: https://w3c.github.io/smufl/gitbook/tables/medieval-and-renaissance-staves.html

fujinaga committed:
New: Add division element to neumes module

The division element is currently missing from the neumes module. These symbols look like barlines but have a function similar to breath marks and rests, with no specific duration.

The lilypond documentation gives some examples of divisions: http://lilypond.org/doc/v2.18/Documentation/notation/typesetting-gregorian-chant#divisiones

They are also available as SMuFL symbols here: https://w3c.github.io/smufl/gitbook/tables/medieval-and-renaissance-staves.html

rettinghaus reviewed (commented):
I'm not against it, but why has this to be a new element? Couldn't rest be used with these additional values? 

And `<memberOf key="att.color"/>` is missing.

ahankinson commented:
They aren't quite rests; they mark out sections as well, and give indicators of phrasing and breaths. So their functions are part rest, part articulation, and part form.

rettinghaus commented:
I'm aware of this, and probably it's best to introduce a new element for this. 
But let's not use a modern english term, but use `<divisio>`(without the "n") instead, which seems to me to be more consistent within this repertoire. Otherwise people may confuse it with the `<div>` division.

ahankinson commented:
The name of the element is consistent with the [names given in the Liber Usualis](http://liber.simssa.ca/#p=13&z=3)

<img width="795" alt="image" src="https://user-images.githubusercontent.com/163183/66523855-cfbb9480-eae8-11e9-9400-12277148e576.png">
<img width="850" alt="image" src="https://user-images.githubusercontent.com/163183/66523886-de09b080-eae8-11e9-80ab-7002dab87324.png">

rettinghaus commented:
That is a non-argument, as the allowed values in type are not `major`, `minor`, `small`, and `final`.

ahankinson commented:
Hm. Perhaps they should be?

rettinghaus commented:
Also I'm strongly against using `@type` to record meaning / visual appearance (see [att.typed](https://music-encoding.org/guidelines/v4/attribute-classes/att.typed.html)). 
I'd prefer `@form` as part of `att.divisio.log`, as we use it for turns and mordents (see [att.turn.log](https://music-encoding.org/guidelines/v4/attribute-classes/att.turn.log.html) and [att.mordent.log](https://music-encoding.org/guidelines/v4/attribute-classes/att.mordent.log.html)).

ahankinson commented:
That's a great suggestion, and I agree

kepper commented:
A lot of discussion has happened in #625, and I'm not sure the Neumes IG has given feedback to that. I believe a consensus is possible, since to me, it seems mostly a discussion on the names of things, but that consensus hasn't been expressed so far. I'm aware that it might be late to get this resolved for today's ODD Friday, but there's a final opportunity on November 29. May I ask @fujinaga, @ahankinson and @rettinghaus to revise the situation? It seems the Neumes IG has done its work, and we should get their changes into the schema now… Thanks everyone :)

martha-thomae commented:
Do you mean October 29th, @kepper ?

martha-thomae commented:
I don't think it is a good idea to call the element "divisio" because in the mensural module we already have a "divisio" attribute. In the case this attribute eventually gets promoted to an element, then we will have some conflicts between both modules.

I just spoke with @fujinaga and he says he doesn't care about the naming ("divisio" or "division").

fujinaga commented:
@kepper So this original #618 can go ahead with the name "division".

craigsapp commented:
> I don't think it is a good idea to call the element "divisio" because in the mensural module we already have a "divisio" attribute. In the case this attribute eventually gets promoted to an element, then we will have some conflicts between both modules.

That is a good reason to avoid "divisio".  Orginally there was a complaint that "division" is English and most terms in the old-music modules were in Latin.  At the neume special interest group in May, a counter-complaint was that using "divisio" is proleptic, meaning that there is no documented name for these symbols, so creating a retrospective Latin name for them would be strange.

Another original complaint is that "division" is close to  div and mdiv, so readability confusion can be caused by using a similar term for something with a very different function.

You might want to consider a more graphical description of the symbols, such as `<vertical-line>` or `<vline>` or `<divLine>`, since it is graphically a vertical line.

To match the symmetry with `<barLine>`, I would suggest the last one: `<divLine>`.  This unites the strong graphical implication of such an element with its function.

craigsapp commented:
> > I just spoke with @fujinaga and he says he doesn't care about the naming ("divisio" or "division").

I don't think so:

> @kepper So this original #618 can go ahead with the name "division".

But you should wait to discuss the pluses/minuses of `<division>` versus `<divLine>` first.

fujinaga commented:
I'm fine with <divLine>.

martha-thomae commented:
> I'm fine with .

martha-thomae commented:
Sorry for closing, it was an accidental click.

craigsapp commented:
<img width="348" alt="Screen Shot 2020-10-22 at 12 57 47 PM" src="https://user-images.githubusercontent.com/3487289/96923481-339bcb80-1466-11eb-8db5-eef6d7ca9240.png">

You need to escape the angle brackets, otherwise no one knows if you voted for `<division>`, `<divisio>` or `<divLine>` 😉 

To do that add backticks around the element name in brackets:

```
`<element>`
```

fujinaga commented:
Sorry, I’m fine with divLine.

Ich

On Oct 22, 2020, at 15:58, Craig Stuart Sapp <notifications@github.com<mailto:notifications@github.com>> wrote:


[Screen Shot 2020-10-22 at 12 57 47 PM]<https://user-images.githubusercontent.com/3487289/96923481-339bcb80-1466-11eb-8db5-eef6d7ca9240.png>

You need to escape the angle brackets, otherwise no one knows if you voted for <division>, <divisio> or <divLine> 😉

To do that add backticks around the element name in brackets:

`<element>`


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/music-encoding/music-encoding/pull/618#issuecomment-714727398>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAH342OUDJQKBFI5UKCQAOLSMCFHDANCNFSM4IZEIZLA>.

craigsapp commented:
This element would also be useful to extend into the CMN module, for cases where the music has barlines, but they do not imply meter.  A good example of this is Charles Ives' second piano sonata:

https://petruccimusiclibrary.ca/files/imglnks/caimg/9/9f/IMSLP03928-Ives_-_Piano_Sonata_No2_%27Concord%27.pdf

Lots of places, but here is a specific example from page 55 of the PDF (or page 57 of the printed edition):

<img width="777" alt="Screen Shot 2020-10-22 at 1 08 06 PM" src="https://user-images.githubusercontent.com/3487289/96924610-bbcea080-1467-11eb-8aa6-1c03704e30b4.png">

In this case the barlines are functioning as phrasing slurs, similar to the vertical lines in chant notation, and there is no implied metricness to these lines.

kepper commented:
The argument to not use "divisio" is compelling. The decision wether it should be called "divLine" or "division" should be taken by the IG. Just for the record: I have some sympathy for "divLine", but let's see what the discussion during today's ODD meeting will bring…

kepper reviewed (approved):
the only thing that's left to decided seems the name, with the two options "division" and "divLine" being under consideration. That should be decided (and fixed, if necessary) today, and then things can be merged.

kepper commented:
Hi @fujinaga, during today's ODD Thursday, we decided to go with `<divLine>`(thanks @craigsapp for that proposal). Could you please update your PR accordingly? In addition, could you rename the attribute `@type` to `@form`, as the first is a generic mechanism in MEI, and we try to keep it free from specific meanings in different contexts (not always to great successful). Also, please add membership to the `att.color` class… 

fujinaga committed:
Update MEI.neumes.xml to include "divLine" element

Update Mei.neumes.xml to include `<divLine>` (aka division or divisio) element.

kepper commented:
Hi @fujinaga, thanks for addressing this so swiftly. Two more minor queries, though: Could you please sort the class memberships alphabetically, and also the values of the `@form` attribute? Thanks very much :-)

fujinaga committed:
Update MEI.neumes.xml for `<divLine>`

Update MEI.neumes.xml. Alphabetized the order of classes and attribute values in `<divLine>`.

kepper reviewed (approved):
wonderful 👍

kepper commented:
thanks to @fujinaga and the whole IG for the continued work on this, and the endurance to get it into the specs ;-)

fujinaga commented:
Thank you!

> On Oct 29, 2020, at 9:16 AM, kepper <notifications@github.com> wrote:
> 
> 
> thanks to @fujinaga and the whole IG for the continued work on this, and the endurance to get it into the specs ;-)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

craigsapp commented:
🥂
